Sources/GRPC/ReadWriteStates.swift

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import NIO
+import SwiftProtobuf
+
+/// Number of messages expected on a stream.
+enum MessageCount {
+  case one
```

Comment:
Should we add `non` and transition into that once a unary call has received its first message? (This may be a dumb idea.)

Reading further, this might replace `canWrite/canRead`, but no strong feelings what is better.